changelog/2024-11-16T11_59_31+01_00_avoid_using_unsafecoerce

```diff
@@ -0,0 +1,2 @@
+CHANGED: Several primitives in `Clash.Promoted.Nat` and `Clash.Sized.Vector` have been rephrased such that they avoid the usage of `unsafeCoerce`, where possible. The changes are only applied for `base >= 4.16.0.0` / `GHC >= 9.2`, as the use of `unsafeCoerce` cannot be avoided with lower versions, unfortunately.
```

Comment:
In what sense is this user-visible? For example, did something crash before that now works? If the users do not notice anything different at all, I think a Changelog entry is unnecessary. If something did change for them, perhaps the entry could be expanded to note what is changed.

Reply:
Also, I think _`base >= 4.16.0.0` / `GHC >= 9.2`_ is a tautology, I'd prefer it just says `GHC >= 9.2`.

Reply:
I removed the changelog entry, as the PR now contains no user-visible changes any more.